pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -133,20 +133,33 @@ func MakeVirtualServices(ing *v1alpha1.Ingress, gateways map[v1alpha1.IngressVis
 }
 
 func makeVirtualServiceSpec(ing *v1alpha1.Ingress, gateways map[v1alpha1.IngressVisibility]sets.String, hosts sets.String) *istiov1alpha3.VirtualService {
-	gw := sets.String{}.Union(gateways[v1alpha1.IngressVisibilityClusterLocal]).Union(gateways[v1alpha1.IngressVisibilityExternalIP])
+	visibility := ing.Spec.Visibility
```

Comment:
Please remove this field which is unused now.

Reply:
Oops!